pkg/plugins/policies/core/rules/rules.go

```diff
@@ -417,16 +419,42 @@ func BuildRules(list []PolicyItemWithMeta) (Rules, error) {
 		return rules, nil
 	}
 
+	uniqueKeys := map[string]struct{}{}
 	// 1. Convert list of rules into the list of subsets
 	var subsets []subsetutils.Subset
 	for _, item := range oldKindsItems {
 		ss, err := asSubset(item.GetTargetRef())
 		if err != nil {
 			return nil, err
 		}
+		for _, tag := range ss {
+			uniqueKeys[tag.Key] = struct{}{}
+		}
 		subsets = append(subsets, ss)
 	}
 
+	// we don't need to generate all permutations when there is no negations
+	// and we have only 0 or one tag, in other cases we need to generate.
+	// in case of `to` policies it can happen when using MeshHTTPRoute
+	if !withNegations && len(uniqueKeys) <= 1 {
```

Comment:
Why do we need to check `uniqueKeys`? I think for `to` policies it should work even if you add route hash tag

Reply:
You are right it works, we don't need it